### ISSUE TITLE
1.12.1 crash with TOP + clipboard

### ISSUE THREAD

**Issue by tyler489** [Nuchaz/BiblioCraft#510]
https://pastebin.com/FMvN9AHC

**Comment by erp-lsf** [comment]
Can vote for this issue too - happens to me in ATM3, on 1.12.

**Comment by croftyraider** [comment]
Same. I came back from caving, opened up my inventory GUI and crashed:

`-- Head --
Thread: Client thread
Stacktrace:
	at net.minecraftforge.client.model.obj.OBJModel$OBJBakedModel.buildQuads(OBJModel.java:1345)
	at net.minecraftforge.client.model.obj.OBJModel$OBJBakedModel.func_188616_a(OBJModel.java:1314)
	at jds.bibliocraft.models.ModelClipboard.func_188616_a(ModelClipboard.java:332)
	at net.minecraft.client.renderer.RenderItem.func_191967_a(RenderItem.java:125)
	at net.minecraft.client.renderer.RenderItem.func_191961_a(RenderItem.java:106)

-- Ingredient being rendered --
Details:
	Ingredient Mod: BiblioCraft
	Ingredient Info: 1xtile.BiblioCraft:Clipboard@0 bibliocraft:clipboard
Stacktrace:
	at mezz.jei.render.IngredientRenderer.createRenderIngredientException(IngredientRenderer.java:216)
	at mezz.jei.render.ItemStackFastRenderer.renderItemAndEffectIntoGUI(ItemStackFastRenderer.java:32)
	at mezz.jei.render.IngredientListBatchRenderer.render(IngredientListBatchRenderer.java:162)
	at mezz.jei.gui.overlay.IngredientGrid.draw(IngredientGrid.java:88)
	at mezz.jei.gui.overlay.IngredientListOverlay.drawScreen(IngredientListOverlay.java:183)
	at mezz.jei.gui.GuiEventHandler.onDrawBackgroundEventPost(GuiEventHandler.java:72)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_1415_GuiEventHandler_onDrawBackgroundEventPost_BackgroundDrawnEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:179)
	at net.minecraft.client.gui.GuiScreen.func_146276_q_(GuiScreen.java:575)
	at net.minecraft.client.gui.inventory.GuiInventory.func_73863_a(SourceFile:66)
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:353)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1124)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1117)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:236)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:297)
	at org.multimc.EntryPoint.listen(EntryPoint.java:162)
	at org.multimc.EntryPoint.main(EntryPoint.java:53)
`

**Comment by arlojeremy** [comment]
Can confirm.

theoneprobe-1.12-1.4.19.jar
BiblioCraft[v2.4.3][MC1.12.0].jar

Moved the mouse down while connecting to the server to keep The One Probe from triggering, and it loaded fine. The instant I pointed The One Probe at the clipboard it crashed.

**Comment by shortybsd** [comment]
Confirming this is happening to another friend on my server. Every time she joins she crashes with
`java.lang.NullPointerException: Rendering ingredient
    at net.minecraftforge.client.model.obj.OBJModel$OBJBakedModel.buildQuads(OBJModel.java:1345)
    at net.minecraftforge.client.model.obj.OBJModel$OBJBakedModel.func_188616_a(OBJModel.java:1314)
    at jds.bibliocraft.models.ModelClipboard.func_188616_a(ModelClipboard.java:332)`

**Comment by joakime** [comment]
Possible duplicate of #490 ?

**Comment by Nuchaz** [comment]
I am still trying to fix this. I've finally been able to reproduce the issue as of today. That is something that I haven't been able to do all this time. I think #490 #535 #537 #540 #542 are all related. Now that I got it crashing for me I an start trying fixes.

**Comment by Nuchaz** [comment]
I think I figured out a fix for this. Expect a fix in the next release, which will be v2.4.4 for MC 1.12.2. I am closing this issue for now. Let me know if anyone still has issues after the next release.

**Comment by Rsslone** [comment]
You're a saint! You waiting on getting other features done before putting out 2.4.4?

**Comment by Nuchaz** [comment]
There is at least 1 or 2 other crashes I am trying to fix first, but I am planning to do a release soon. This weekend at the latest I think. Maybe tomorrow at the earliest. Ideally I want to pack in as many fixes as I can, but now that I've finally made some headway on this crash I am anxious to get a version out there.

**Comment by Rsslone** [comment]
Sounds great. Just in time for my modpack release. Let me tell ya, the clipboard is the most important part of an industrial factory. :P
